.mbedignore

```diff
@@ -6,7 +6,21 @@ azure-iot-sdk-c/iothub_client/src/iothub_client_ll_uploadtoblob.c
 azure-iot-sdk-c/iothub_client/src/iothubtransportmqtt_websockets.c
 azure-iot-sdk-c/iothub_client/tests/*
 azure-iot-sdk-c/iothub_service_client/*
-azure-iot-sdk-c/provisioning_client/*
+azure-iot-sdk-c/provisioning_client/adapters/hsm_client_http_edge.*
+azure-iot-sdk-c/provisioning_client/adapters/hsm_client_riot.*
+azure-iot-sdk-c/provisioning_client/adapters/hsm_client_tpm.*
+azure-iot-sdk-c/provisioning_client/samples/*
+azure-iot-sdk-c/provisioning_client/tests/*
+azure-iot-sdk-c/provisioning_client/tools/*
+azure-iot-sdk-c/provisioning_client/inc/azure_prov_client/internal/prov_transport_amqp_common.h
+azure-iot-sdk-c/provisioning_client/inc/azure_prov_client/internal/prov_sasl_tpm.h
```

Comment:
This doesn't have any effect, because compilers only care about include paths ("internal/" in this case), not individual header files.